File: ExpandableHeaders-ObjC/ExpandableHeaders/UIColor+AppColors.m
//
//  UIColor+AppColors.m
// LifeWorks
//
//  Created by James Johnson on 06/01/2016.
//  Copyright © 2016 Anexinet. All rights reserved.
//

#import "UIColor+AppColors.h"

@implementation UIColor (appColors)
// custom color methods
+ (UIColor *)colorWithHexString:(NSString *)str {
    if (str == nil) {
        return [UIColor colorWithHex:(int)0xFF5300];
    } else {
        const char *cStr = [str cStringUsingEncoding:NSASCIIStringEncoding];
        long x = strtol(cStr + 1, NULL, 16);
        
        return [UIColor colorWithHex:(int)x];
    }
}

+ (UIColor *)colorWithHex:(int)hex {
    
    int r = (hex >> 16) & 0xFF;
    int g = (hex >> 8) & 0xFF;
    int b = hex & 0xFF;
    
    return [UIColor colorWithRed:(float)r / 255.0f
                           green:(float)g / 255.0f
                            blue:(float)b / 255.0f
                           alpha:1.0f];
}

+ (void)changeImageColor:(UIImageView *)theImageView toColor:(UIColor *)newColor {
    theImageView.image = [theImageView.image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
    theImageView.tintColor = newColor;
}

@end

File: ExpandableHeaders-Swift/ExpandableHeaders/UIColor+AppColors.swift
//
//  UIColor+AppColors.swift
//  ExpandableHeaders
//
//  Created by James Johnson on 05/02/2017.
//  Copyright © 2017 Anexinet. All rights reserved.
//

import UIKit

extension UIColor {

    // custom color methods
    class func colorWithHex(rgbValue: UInt32) -> UIColor {
        return UIColor( red: CGFloat((rgbValue & 0xFF0000) >> 16) / 255.0,
                      green: CGFloat((rgbValue & 0x00FF00) >> 8) / 255.0,
                       blue: CGFloat(rgbValue & 0x0000FF) / 255.0,
                      alpha: CGFloat(1.0))
    }
    
    class func colorWithHexString(hexStr: String) -> UIColor {
        var cString:String = hexStr.trimmingCharacters(in: .whitespacesAndNewlines).uppercased()

        if (hexStr.hasPrefix("#")) {
            cString.remove(at: cString.startIndex)
        }
        if (cString.isEmpty || (cString.characters.count) != 6) {
            return colorWithHex(rgbValue: 0xFF5300);
        } else {
            var rgbValue:UInt32 = 0
            Scanner(string: cString).scanHexInt32(&rgbValue)
            
            return colorWithHex(rgbValue: rgbValue);
        }
    }
    
    func changeImageColor(theImageView: UIImageView, newColor: UIColor) {
            theImageView.image = theImageView.image?.withRenderingMode(.alwaysOriginal)
            theImageView.tintColor = newColor;
    }
}
